pkg/connection/config.go

```diff
@@ -31,13 +31,39 @@ type Config struct {
 	Token    string `mapstructure:"token"`    // bearer token
 }
 
+// OverrideActiveConfigRegistryProject forces the active configuration to use the specified project.
+func OverrideActiveConfigRegistryProject(project string) {
+	// force the config to be read and cached
+	useCache = true
+	_, err := ActiveConfig()
+	if err == nil && cachedConfig != nil {
+		cachedConfig.Project = project
+	}
+}
+
+var cachedConfig *Config
+var useCache bool
```

Comment:
These globals should probably be at the top of the file. I don't think I'm particularly unusual with that being the first place I look when functions access a variable that isn't a parameter.

Also, do we need globals here? I'm not familiar enough with the code to propose other designs, but maybe there are other more functional options that are less likely to cause surprises in the future.

Reply:
I put those variables there to be close to their point of use... I don't mind moving them but I'm unsure that it's helpful. We should probably just get rid of them...

if Go supported static variables inside functions, I probably would have used that instead. [This](https://stackoverflow.com/a/30558173) might be a reasonable alternative.